# CWL and Dockerfile for running postprocessing_variant_calls

## Version of tools in docker image (/container/Dockerfile)

| Tool	| Version	| Location	|
|---	|---	|---	|
| ubuntu  	| 18.04  	|  -	|
| postprocessing_variant_calls  	| 0.0.1 	|  https://github.com/msk-access/postprocessing_variant_calls	|


## CWL case-control_filter_0.0.1.cwl

- CWL specification 1.0
- Use example_inputs_case-control_filter.yaml to see the inputs to the cwl
- Example Command using [toil](https://toil.readthedocs.io):

```bash
    > toil-cwl-runner case-control_filter_0.0.1.cwl example_inputs_case-control_filter.yaml
```
**If at MSK, using the JUNO cluster having installed toil-msk version 3.21.1 you can use the following command**

```bash
#Using CWLTOOL
> cwltool --singularity --non-strict /path/to/case-control_filter_0.0.1.cwl /path/to/inputs.yaml

#Using toil-cwl-runner
> toil-cwl-runner --singularity --logFile /path/to/case-control_filter_0.0.1_toil.log  --jobStore /path/to/case-control_filter_0.0.1_jobStore --batchSystem lsf --workDir /path/to/case-control_filter_0.0.1_toil_log --outdir . --writeLogs /path/to/case-control_filter_0.0.1_toil_log --logLevel DEBUG --stats --retryCount 2 --disableCaching --disableChaining --maxLogFileSize 20000000000 /path/to/case-control_filter_0.0.1.cwl /path/to/inputs.yaml > case-control_filter_0.0.1_toil.stdout 2> case-control_filter_0.0.1_toil.stderr &
```

### Usage: 

```
Usage: pv vardict case-control filter [OPTIONS]

  This tool helps to filter vardict version 1.4.6 VCFs for case control
  calling

Options:
  -i, --inputVcf FILE             Input vcf generated by vardict which needs
                                  to be processed  [required]
  --tsampleName TEXT              Name of the tumor Sample  [required]
  -dp, --totalDepth INTEGER RANGE
                                  Tumor total depth threshold  [default: 20;
                                  x>=20]
  -ad, --alleledepth INTEGER RANGE
                                  [x>=1]
  -tnr, --tnRatio INTEGER         Tumor-Normal variant fraction ratio
                                  threshold  [default: 1]
  -vf, --variantFraction FLOAT    Tumor variant fraction threshold  [default:
                                  5e-05]
  -mq, --minQual INTEGER          Minimum variant call quality  [default: 0]
  -fg, --filterGermline           Whether to remove calls without 'somatic'
                                  status
  -o, --outDir TEXT               Full Path to the output dir
  --help                          Show this message and exit.
``` 

## CWL single_filter_0.0.1
- CWL specification 1.0
- Use example_inputs_single_filter.yaml to see the inputs to the cwl
- Example Command using [toil](https://toil.readthedocs.io):

```bash
    > toil-cwl-runner single_filter_0.0.1 example_inputs_single_filter.yaml
```
**If at MSK, using the JUNO cluster having installed toil-msk version 3.21.1 you can use the following command**

```bash
#Using CWLTOOL
> cwltool --singularity --non-strict /path/to/single_filter_0.0.1 /path/to/inputs.yaml
#Using toil-cwl-runner
> toil-cwl-runner --singularity --logFile /path/to/single_filter_0.0.1_toil.log  --jobStore /path/to/single_filter_0.0.1_jobStore --batchSystem lsf --workDir /path/to/single_filter_0.0.1_toil_log --outdir . --writeLogs /path/to/single_filter_0.0.1_toil_log --logLevel DEBUG --stats --retryCount 2 --disableCaching --disableChaining --maxLogFileSize 20000000000 /path/to/single_filter_0.0.1_0.7.17.cwl /path/to/inputs.yaml > single_filter_0.0.1_toil.stdout 2> single_filter_0.0.1_toil.stderr &
```

### Usage 

```
Usage: pv vardict single filter [OPTIONS]

  This tool helps to filter vardict version 1.4.6 VCFs for single sample
  calling

Options:
  -i, --inputVcf FILE             Input vcf generated by vardict which needs
                                  to be processed  [required]
  --tsampleName TEXT              Name of the tumor Sample  [required]
  -dp, --totalDepth INTEGER RANGE
                                  Tumor total depth threshold  [default: 20;
                                  x>=20]
  -ad, --alleledepth INTEGER RANGE
                                  [x>=1]
  -tnr, --tnRatio INTEGER         Tumor-Normal variant fraction ratio
                                  threshold  [default: 1]
  -vf, --variantFraction FLOAT    Tumor variant fraction threshold  [default:
                                  5e-05]
  -mq, --minQual INTEGER          Minimum variant call quality  [default: 0]
  -fg, --filterGermline           Whether to remove calls without 'somatic'
                                  status
  -o, --outDir TEXT               Full Path to the output dir
  --help                          Show this message and exit.
``` 
